main.py

```diff
@@ -109,79 +169,139 @@ def resize_app(self) -> App:
+        # TODO: Give all data in dual columns
+        # | Weather: Clear    | Temp: 20°C |
+        # TODO: Allow users to change between imperial and metric (Fahrenheit and Celsius)
         data = response.json()
         main = data["main"]
         temperature = weather.temperature("celsius")
-        wind = weather.wind(unit="meters_sec")
-        # Get temperature in Celsius
-        # temperature_kelvin: float = main["temp"]
-        # temperature_celsius = temperature_kelvin - 273.15
-        temp = temperature.get("temp", None)
-        temp_max = temperature.get("temp_max", None)
-        temp_min = temperature.get("temp_min", None)
-        feels_like = temperature.get("feels_like", None)
-        status: str = weather.status
-        detailed_status: str = weather.detailed_status
-        # Other data needed from the API
-        humidity = main["humidity"]
-        pressure = main["pressure"]
-        visibility = weather.visibility(unit="kilometers")
-        windspeed: float = wind["speed"]
-        # Put in label
-        self.cityname.configure(
-            text=data["name"] + ", " + data["sys"]["country"])
-        self.label_status.configure(
-            text="Weather: " + status + ":- " + detailed_status)
-        self.label_temp.configure(text="Temperature: " + f"{temp:.2f}°C")
-        self.label_temp_max.configure(
-            text="Maximum Temperature: " + f"{temp_max:.2f}°C"
-        )
-        self.label_temp_min.configure(
-            text="Minimum Temperature: " + f"{temp_min:.2f}°C"
-        )
-        self.label_feels_like.configure(
-            text="Feels like " + f"{feels_like:.2f}°C")
-        self.label_humidity.configure(text="Humidity: " + f"{humidity:.2f}%")
-        self.label_pressure.configure(
-            text="Pressure: " + f"{pressure:.2f}" + " hPa")
-        self.label_visibility.configure(
-            text="Visibility: " + f"{visibility:.2f}" + " km"
-        )
-        self.label_windspeed.configure(
-            text="Wind Speed: " + f"{windspeed:.2f}" + " meters per second"
+
+        # Update labels
+        self.update_labels(
+            [
+                f"Weather: {weather.status} ~ {weather.detailed_status}",
+                f"Current Temperature: {temperature.get('temp', None):.2f}°C",
+                f"Maximum Temperature: {temperature.get('temp_max', None):.2f}°C",
+                f"Minimum Temperature: {temperature.get('temp_min', None):.2f}°C",
+                f"Feels like {temperature.get('feels_like', None):.2f}°C",
+                f"Humidity: {main['humidity']:.2f}%",
+                f"Pressure: {main['pressure']:.2f} hPa",
+                f"Visibility: {weather.visibility(unit='kilometers'):.2f} km",
+                f"Wind Speed: { weather.wind(unit='meters_sec')['speed']:.2f} meters per second",
+            ]
         )
-        return self
+
+        # Set the city name
+        self.cityname.configure(text=f"City: {city}")
```

Comment:
Would you please put it all in the f-string?

Reply:
Well, isn't it already in a f-string? What are you referring to?

Reply:
No need for a variable is what I mean.

Reply:
Oh, now I get it. I'll fix it, I need some time. It's already nearing midnight and I'm doing homework. The Life of A Teenage Coder?

I will commit it, no problem.

Reply:
Sounds good! Make sure to finish that homework!

Reply:
Well, two things. I did my homework, and the change is already made. You added it, like you said. I recommended this before that commit, and now it's outdated. So, don't worry much, you are good to merge this PR. But can you try placing the drop-down menus for temperature and color below the city heading? It kinda feels a bit cramped, but what is your say on this?

Reply:
I tried about multiple different designs and asked friends to rank each and this one was rated highest :(.

Reply:
I agree it doesn't look perfect but couldn't find a better design either.

Reply:
Well, for now, this would be fine, but we eventually have to add a settings window. Trust me, people who want to use their own APIs here can do so, but we can make it easier by adding GUI options for that, and more. I would be getting ideas on this from now on.

A quick alert: I won't be as active as I am after September. My school is planning on more prolonged classes, etc. and etc. There is a guaranteed load of homework, so I won't code much, and that might as well include the holidays. So I am just telling that here, to alert you all.

Reply:
Where possible, don't use popups or extra windows. I'll figure out a better design eventually.

Reply:
We'll use 2 windows. One for the main app, and one for the unified settings, about page and update page, all tabbed. What would your idea be?

P.S. gotta move this to an issue soon

Reply:
If you insist on a second window, I guess that can work but will advise against it.

Reply:
It already is. It could use a better design, sure, but a whole window for two settings? That's a little over the top.

Reply:
We could repurpose the same window for settings too. Any ideas on this type? We could use a tabbed window. I'll explain.